### PR TITLE
Correct reading HTTP client response body without content type + `ByteBuffer` converters

### PR DESCRIPTION
**Pull request by dstepanov**
The buffer of the client response should be discarded

### CODE REVIEW

File: buffer-netty/src/main/java/io/micronaut/buffer/netty/NettyByteBufferConverters.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        conversionService.addConverter(
+            ByteBuf.class,
+            ByteBuffer.class,
+            NettyByteBuffer::new
+        );
+
+        conversionService.addConverter(
+            java.nio.ByteBuffer.class,
+            ByteBuf.class,
+            nioBuffer -> Unpooled.copiedBuffer(nioBuffer)
```
Comment (yawkat) [inline buffer-netty/src/main/java/io/micronaut/buffer/netty/NettyByteBufferConverters.java:55]: why copy and not wrap?

File: http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java [hunk 2]
```diff
@@ -562,18 +562,9 @@ public <I, O, E> io.micronaut.http.HttpResponse<O> exchange(io.micronaut.http.Ht
...
-                Flux<HttpResponse<O>> publisher = Flux.from(DefaultHttpClient.this.exchange(request, bodyType, errorType, blockHint));
-                return publisher.doOnNext(res -> {
-                    Optional<ByteBuf> byteBuf = res.getBody(ByteBuf.class);
-                    byteBuf.ifPresent(bb -> {
-                        if (bb.refCnt() > 0) {
-                            ReferenceCountUtil.safeRelease(bb);
-                        }
-                    });
-                    if (res instanceof FullNettyClientHttpResponse response) {
-                        response.onComplete();
```
Comment (yawkat) [inline http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java:574]: are you sure we don't need the onComplete here

File: http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java [hunk 3]
```diff
@@ -1447,7 +1432,6 @@ public void onComplete() {
                                     try {
                                         FullHttpResponse fullHttpResponse = new DefaultFullHttpResponse(nettyResponse.protocolVersion(), nettyResponse.status(), buffer, nettyResponse.headers(), new DefaultHttpHeaders(true));
                                         final FullNettyClientHttpResponse<Object> fullNettyClientHttpResponse = new FullNettyClientHttpResponse<>(fullHttpResponse, handlerRegistry, (Argument<Object>) errorType, true, conversionService);
-                                        fullNettyClientHttpResponse.onComplete();
```
Comment (yawkat) [inline http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java:1450]: same here

File: http-client/src/main/java/io/micronaut/http/client/netty/FullNettyClientHttpResponse.java [hunk 4]
```diff
@@ -221,19 +220,14 @@ private <T> Optional<T> convertByteBuf(ByteBuf content, Argument<T> type) {
...
         } else if (LOG.isTraceEnabled()) {
             LOG.trace("Missing or unknown Content-Type received from server.");
         }
         // last chance, try type conversion
-        return conversionService.convert(content, ByteBuf.class, type);
-    }
-
-    @Override
-    public void onComplete() {
-        this.complete = true;
```
Comment (yawkat) [inline http-client/src/main/java/io/micronaut/http/client/netty/FullNettyClientHttpResponse.java:236]: oh, complete was unused? :smile:

**Review by yawkat — APPROVED** [review]

File: http-client/src/main/java/io/micronaut/http/client/netty/FullNettyClientHttpResponse.java (hunk 4 above)
Comment (dstepanov) [reply http-client/src/main/java/io/micronaut/http/client/netty/FullNettyClientHttpResponse.java:236]: Yeah, the field was highlighted as not used

**Review by graemerocher — APPROVED** [review]

**Pull request merged**